src/Console/Command/Git/InitCommand.php

```diff
@@ -133,7 +133,8 @@ protected function parseHookBody(string $hook, string $templateFile): string
         ];
 
         foreach ($this->hooksConfig->getVariables() as $key => $value) {
-            $replacements[sprintf('$(%s)', $key)] = ProcessUtils::escapeArgumentsFromString($value);
+            $process = is_array($value) ? new Process($value) : Process::fromShellCommandline($value);
```

Comment:
Wouldn't it be better to add a configuration normalizer that converts a string into an array?
That way we can always assume we have an array, even if the `grumphp.yaml` contains a string.

I am not sure how this solves your issue though: it still escapes the configured CLI argument.
If you are on windows, it escapes with double quotes, so it might work.
However, if you are on linux: it escapes with single quotes, meaning it won't work.

Not sure if it's a good idea, but maybe we shouldn't escape the variables and leave that up to the one configurating grumphp? That way you can do something like:


```yaml
grumphp:
    git_hook_variables:
        EXEC_GRUMPHP_COMMAND: ['docker-composer --rm --no-deps -u "$(id -u):$(id -g)"', "'php'" ]
```

Which will result in:

```
docker-composer --rm --no-deps -u "$(id -u):$(id -g)" 'php'
```


Meaning you can choose if you want to escape or not from the config file.
However: it means we won't escape by default anymore.... 

(Assuming it is a developer tool, this won't be a big security issue IMO.)

Reply:
In this way it's up to the developer if the command will be escaped or not. If someone uses a string it won't be escaped automatically, but if someone uses an array it will be escaped by the process component.
Won't doing automatic escaping would be the other option, but I think it's more convenient this way. It's up to the user to decide.

If it would be normalized to an array how could we decide if it needs escaping or not?

In my case I can use `EXEC_GRUMPHP_COMMAND: 'docker-compose run --rm --no-deps -u $(id -u):$(id -g) php'`

Reply:
sounds good!